ext/bcmath/libbcmath/src/str2num.c

```diff
@@ -57,77 +60,91 @@ bool bc_str2num(bc_num *num, char *str, size_t scale)
 	while (*ptr == '0') {
 		ptr++;
 	}
+	integer_ptr = ptr;
 	/* digits before the decimal point */
 	while (*ptr >= '0' && *ptr <= '9') {
 		ptr++;
 		digits++;
 	}
 	/* decimal point */
-	if (*ptr == '.') {
-		ptr++;
+	decimal_point = (*ptr == '.') ? ptr : NULL;
+
+	/* If a string other than numbers exists */
```

Comment:
I think this is clearer, although probably not the ideal wording either:
```suggestion
	/* If a non-digit and non-decimal-point indicator is in the string, i.e. an invalid character */
```